`silu`: port to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58050 Port silu to structured**
* #57623 Port softshrink to structured
* #57622 Port hardsigmoid to structured
* #57621 Port leaky_relu to structured
* #57620 Port softplus to structured
* #57619 Port elu to structured

Differential Revision: [D28382790](https://our.internmc.facebook.com/intern/diff/D28382790)

Freey0 committed:
Port silu to structured

[ghstack-poisoned]

Freey0 committed:
Update on "Port silu to structured"




[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/Activation.cpp:
```diff
@@ -252,25 +262,6 @@ Tensor & celu_(Tensor & self, const Scalar& alpha) {
-Tensor& silu_out(const Tensor& self, Tensor& result) {
-  TORCH_CHECK(
-      result.dtype() == self.dtype(),
-      "Output Tensor should have the same type as in Input Tensor.")
```
hmmmmm I wonder what's going on with this check

Freey0 replied:
This check is not needed, because it is already checked in build_unary_op that the input and output have the same type.
```
void TensorIteratorBase::build_unary_op(const Tensor& out, const Tensor& a) {
  build(TensorIteratorConfig()
      .set_check_mem_overlap(true)
      .add_output(out)
      .add_input(a)
      .cast_common_dtype_to_outputs(false)
      .enforce_safe_casting_to_output(false)
      .check_all_same_dtype(true));    <--------HERE
}
```
Older code shouldn't need this check either

ezyang commented:
but need to fix buildf ailures

Freey0 committed:
Update on "Port silu to structured"




[ghstack-poisoned]

Freey0 committed:
Update on "Port silu to structured"




[ghstack-poisoned]

Freey0 committed:
Update on "Port silu to structured"




[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28382790).

ezyang commented:
@Freey0 looks like there is merge conflict on this stack, can you rebase to latest master?

Freey0 committed:
Update on "Port silu to structured"


Differential Revision: [D28382790](https://our.internmc.facebook.com/intern/diff/D28382790)

[ghstack-poisoned]

Freey0 commented:
> @Freey0 looks like there is merge conflict on this stack, can you rebase to latest master?

Sure, done!

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28382790).
